frontend/src/components/MapView/Legends/LegendItem/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React, {
+  memo,
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import {
+  Box,
+  createStyles,
+  Divider,
+  Grid,
+  ListItem,
+  Paper,
+  Slider,
+  Typography,
+  withStyles,
+  WithStyles,
+} from '@material-ui/core';
+import { useDispatch, useSelector } from 'react-redux';
+import { LayerType, LegendDefinitionItem } from '../../../../config/types';
+import { mapSelector } from '../../../../context/mapStateSlice/selectors';
+import { useSafeTranslation } from '../../../../i18n';
+import { setAnalysisLayerOpacity } from '../../../../context/analysisResultStateSlice';
+import LayerContentPreview from '../layerContentPreview';
+import { handleChangeOpacity } from '../handleChangeOpacity';
+import ColorIndicator from '../ColorIndicator';
+import { getLegendItemLabel } from '../../utils';
+import LoadingBar from '../LoadingBar';
+
+// Children here is legendText
+const LegendItem = memo(
+  ({
+    classes,
+    id,
+    title,
+    legend,
+    type,
+    opacity: initialOpacity,
+    children,
+    legendUrl,
+    displayOpacitySlider,
+  }: LegendItemProps) => {
+    const dispatch = useDispatch();
+    const map = useSelector(mapSelector);
+    const [opacity, setOpacityValue] = useState<number | number[]>(
+      initialOpacity || 0,
+    );
+
+    useEffect(() => {
+      setOpacityValue(initialOpacity || 0);
+    }, [initialOpacity]);
+
+    const { t } = useSafeTranslation();
+
+    const handleChangeOpacityValue = useCallback(
+      val => {
+        setOpacityValue(val);
+        dispatch(setAnalysisLayerOpacity(val));
+      },
+      [dispatch],
+    );
+
+    const renderedOpacitySlider = useMemo(() => {
+      if (!displayOpacitySlider) {
+        return null;
+      }
+      return (
+        <Grid item className={classes.slider}>
+          <Box px={1}>
+            <Slider
+              value={opacity}
+              step={0.01}
+              min={0}
+              max={1}
+              aria-labelledby="opacity-slider"
+              onChange={(e, newValue) =>
+                handleChangeOpacity(
+                  e,
+                  newValue as number,
+                  map,
+                  // TODO - get updated layer id in groups after switch. See issue #743
```

Comment:
@HarryMytilinaios is this resolved?

Reply:
Yes @ericboucher I think this is resolved I am removing the TODO.